src/annotator/index.js

```diff
@@ -34,32 +34,49 @@ const appLinkEl = /** @type {Element} */ (
 );
 
 function init() {
+  window_.__hypothesis = {};
+
   const annotatorConfig = getConfig('annotator');
   const isPDF = typeof window_.PDFViewerApplication !== 'undefined';
 
-  if (annotatorConfig.subFrameIdentifier) {
-    // Other modules use this to detect if this
-    // frame context belongs to hypothesis.
-    // Needs to be a global property that's set.
-    window_.__hypothesis_frame = true;
-  }
-
   const eventBus = new EventBus();
   const guest = new Guest(document.body, eventBus, {
     ...annotatorConfig,
     // Load the PDF anchoring/metadata integration.
     // nb. documentType is an internal config property only
     documentType: isPDF ? 'pdf' : 'html',
   });
+
```

Comment:
Perhaps a comment explaining that the host will create the sidebar, while the guest-only frames (which have a `subFrameIdentifier`)...don't. Anything to help reduce the magic-ness of `subFrameIdentifier` here would be helpful!